sql/codegen_alps.go

```diff
@@ -429,7 +429,8 @@ func newALPSTrainFiller(pr *extendedSelect) (*alpsFiller, error) {
 	}
 	modelDir := fmt.Sprintf("%s/model/", scratchDir)
 
-	fcList, fsMap, err := resolveTrainColumns(&pr.columns)
+	cols := pr.cc[""]
+	fcList, fsMap, err := resolveTrainColumns(&cols)
```

Comment:
Why do we need the assignment to `cols`? Can we have just one line as the following?

```go
	fcList, fsMap, err := resolveTrainColumns(&(pr.cc[""]))
```

There are a few more cases below.

Reply:
GO forbid to take the address of a `map` member because of the address may be invalid if the map grows or shrinks.